app/models/piece.rb

```diff
@@ -11,4 +11,42 @@ class Piece < ActiveRecord::Base
   scope :queens, -> {where(type: 'Queen') }
   scope :pawns, -> {where(type: 'Pawn') }
 
+  # define some helper methods for move validation
+  def raise_exceptions(position_2)
+    raise "this move is not within the board" unless self.move_within_board?(position_2)
+    raise "this move is obstructed" if (self.game.is_obstructed?(self.current_position, position_2) && !(self.instance_of? Knight))
```

Comment:
I wouldn't raise an exception in this case, since it's not an exceptional case, it's just an invalid move.  Instead, I would go down and make `valid_move?` return false in this case.